adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/MessageMapping.java

```diff
@@ -87,7 +86,7 @@ public void setWebClient(final WebClient webClient) {
     Future<MappedMessage> mapMessageRequest(final MqttContext ctx, final ResourceIdentifier targetAddress,
                                             final MqttPublishMessage message, final MapperEndpoint mapperEndpoint, final VertxHttpHeaders headers) {
         final Promise<MappedMessage> result = Promise.promise();
-        getWebClient().post(mapperEndpoint.getPort(), mapperEndpoint.getHost(), mapperEndpoint.getUri())
+        webClient.post(mapperEndpoint.getPort(), mapperEndpoint.getHost(), mapperEndpoint.getUri())
```

Comment:
it looks like the result will never be completed if the HTTP request fails ...

Reply:
good catch. It should now